.gitbook/3-protocol-devs/1-dep-specs/4-data-validation.md

```diff
@@ -0,0 +1,217 @@
+    "deal_id": 1,
+    "provider_address": "panacea1ewugvs354xput6xydl5cd5tvkzcuymkejekwk3",
+    "data_hash": "13341f91f0da76d2adb67b519e2c9759822ceafd193bd26435ba0d5eee4c3a2b"
+  },
+  "signature": "MEQCIEPyGSe9wVIjrUFuzXtQtEc0siwaHkp4QJCMBvC8ttWQAiAUkntIQldgIkIBFdthaTRWXHisDV2Ys/Ufpc9zejUEuQ=="
+}
+```
+
+| Key                                   | Type   | Description                                                      |
+|---------------------------------------|--------|------------------------------------------------------------------|
+| unsigned_certificate                  | Object | Unsigned certificate containing data validation information      |
+| unsigned_certificate.cid              | string | A content identifier of IPFS                                     |
+| unsigned_certificate.unique_id        | string | UniqueID of the oracle that validated the data                   |
+| unsigned_certificate.oracle_address   | string | Account address of the oracle that validated the data            |
+| unsigned_certificate.deal_id          | int    | Deal to whom the provider intends to provide data                |
+| unsigned_certificate.provider_address | string | Data provider's account address                                  |
+| unsigned_certificate.data_hash        | string | The value of hashing the original data with the SHA256 algorithm |
+| signature                             | string | The value of unsigned_certificate signed with OracleKey          |
+
+
+### Data validation process
+
+#### Data Decryption
+Provider's encrypted data(`encrypted_data_base64`) can only be decrypted by Oracle.
+
+TODO: Guide to decrypt data
+```
+secret_key = SHA256(ECDH(oracle_private_key, provider_public_key))
+
+encrypted_data = Base64.Decode(encrypted_data_base64)
+
+orgin_data = AES256GCM.Decrypt(secret_key, encrypted_data)
+```
+
+#### Data Validation
+After hashing the original data with SHA256, it is encoded with HEX and compared with `data_hash`.
+```
+compare(data_hash, HEX.Encode(SHA256(origin_data))
+```
+
+Verify that the `provider_address` of the original data matches the JWT auth token issuer of the request header.
+```
+compare(origin_data.provider_address,jwtToken.issuer)
+```
+
+Deal is retrieved from the Panacea using the dealID got from the URI path.
+```
+deal = getDealFromPanacea(deal_id)
+```
+
+Before verifying the data, it is checked whether the deal status is valid.
+If the Deal's status is invalid, Oracle does not perform verification work.
+```
+compare(deal.status, 'DEAL_STATUS_ACTIVE')
+```
+
+Validate original data with `data_schema` extracted from Deal. 
+We currently support JSON schema.
+```
+data_schema = deal.data_schema
+
+JSONSchema.Validate(data_schema, orgin_data)
+```
+
+#### Data re-encryption and store to IPFS
+
+If data validation is successful, the data must be re-encrypted and stored on IPFS. 
+
+This encrypted data is stored in IPFS to be delivered to the consumer.
```

Comment:
I think this sentence is not necessary because you are mentioning it at the end of this chapter.

Reply:
I mean the sentence `This encrypted data is stored in IPFS to be delivered to the consumer`.